activities/codebase-stewardship/product-development.md

```diff
@@ -0,0 +1,121 @@
+---
+type: Resource
+---
+
+# Codebase Product Development
+
+## Product life cycle
+
+A [product](../../glossary/product-definition.md) starts with an idea which reflects a need or provides a solution to specific problem. In the context of public code the indented product often supports the management and or execution of public policy.
+
+The process from and initial idea to a fully developed product is an evolutionary process supported by a variety of development methodologies and management frameworks (see further reading). A product is constantly refined, altered and improved. This entire process is often referred to as the product life cycle.
+
+The product life cycle can be described from different perspectives. From an economic perspective such as the [Product Life Cycle Theory](https://en.wikipedia.org/wiki/Product_life-cycle_theory) or from a [product engineering perspective](https://en.wikipedia.org/wiki/Product_lifecycle). Designing, developing and managing a product is a multi-disciplinary activity which touches all types of activities in an organization ranging from marketing to business development to customer support.
+
+From a high-level overview the product life cycle could be divided in the following phases:
+
+* Phase 1: Conceive - Imagine, specify, plan, innovate
+* Phase 2: Design - Describe, define, develop, test, analyze and validate
+* Phase 3: Realize - Manufacture, make, build, procure, produce, sell and deliver
+* Phase 4: Service - Use, operate, maintain, support, sustain, phase-out, retire
+* All phases - Communicate, manage and collaborate
+
+It is imported to note that none of the above phases should be considered as isolated. In reality, a product life cycle does not run sequentially or separated from other product development activities, with information flowing between different people and systems. These phases are constantly revisited and revised during the lifespan of a product. Software development using agile frameworks where these kind of phases are executed iteratively and in parallel.
+
+## Product stewardship activities in the product life cycle
+
+In the context of codebase stewardship the goal of product stewardship is to support codebases and their communities. This helps them build and grow a viable and sustainable ecosystem around a codebase, Then a community can use shared resources and knowledge to support the product life cycle for the duration for its lifespan.
```

Comment:
"its lifespan" - does this refer to the product, the community, or even the product life cycle?